### ISSUE TITLE
Dimensionality Reduction: PCA, tSNE, UMAP

### ISSUE THREAD

**Issue by AnabelSMRuggiero** [rladies-seattle/rladies-seattle-organizing#24]
# Topic

### Who is the audience?
UMAP was a specifically requested topic on the R-Ladies Seattle slack. More generally, dimensionality reduction is a useful technique for data visualization and as an initial layer of a model. This makes it a powerful tool in a data scientist's arsenal.

### Why is this important?
Data visualization: High dimensional data is tricky to present. Dimensionality reduction helps visualize data by projecting the data to 1, 2, or 3 dimensions.
Modeling: Most models will have at least one step involving O(n^2) or greater time with respect to input dimensionality. Reducing the dimensionality of a model input by half reduces the time taken by an O(n^2) operation by 75%. This means faster models and faster fitting. Additionally, dimensionality reduction techniques transform the input to a coordinate system that may prime your model for success.

### What should be covered?
I plan to go over the basics of the math and algorithms behind PCA, tSNE, and UMAP. Most of the time will be devoted to a R example implementing these algorithms.

I have not set my eyes on a specific example, so I'm open if anyone has a specific use case they want covered. 

### Suggested speakers or contributors
I'm expecting to present this myself; I am unsure of who else to suggest but am more than willing to collaborate.

### Resources you would recommend to the audience
Here are some of the pages I stumbled across gathering my thoughts on the topic:
https://umap-learn.readthedocs.io/en/latest/how_umap_works.html
https://arxiv.org/abs/1802.03426
https://towardsdatascience.com/tsne-vs-umap-global-structure-4d8045acba17
https://en.wikipedia.org/wiki/Principal_component_analysis
Matrix Computations by Golub and Van Loan go over the steps behind the SVD, the primary algorithm that's used to implement PCA.

### I'm interested to hear what people want or would expect from this topic.

**Comment by lmcinnes** [comment]
I would be happy to help out in any way I can. You can reach me via email (leland DOT mcinnes AT gmail DOT com), here on github, or on twitter.

**Comment by fedemolina** [comment]
It would be nice to apply this techniques to product characteristics in retail sector. You can use dataset from Investigating Effects of Out-of-Stock on Consumer Stockkeeping Unit Choice, HAI CHE, XINLEI (JACK) CHEN, and YUXIN CHEN, Journal of Marketing Research. 
For example in SHOPPER: A PROBABILISTIC MODEL OF CONSUMER CHOICE WITH SUBSTITUTES AND COMPLEMENTS, of Athey, Ruiz & Blei they used t-sne.

**Comment by endimeon777** [comment]
Best regards

I am trying to make a comparison of the incremental versions of dimensionality reduction methods. Could someone tell me where I can find the code for Incremental Locally Linear Embedding, or Incremental Multidimensional Scaling or Incremental Laplacian EigenMaps ?

Thank you.

**Comment by AnabelSMRuggiero** [comment]
@endimeon777 Unfortunately, incremental dimensionality reduction methods are outside the scope of what I have prepared/am preparing for this tutorial proposal, and not an area I'm familiar with. A cursory google search for implementations has come up empty for implementations, but I did find academic papers on these algorithms. If the articles themselves don't give leads on where to find implementations, it might be worth contacting the author and asking them if it is available.
